lib/vfscore/main.c

```diff
@@ -1167,36 +1182,85 @@ int readdir_r(DIR *dir, struct dirent *entry, struct dirent **result)
 	int error;
 	struct vfscore_file *fp;
 
+	/**
+	 * `sys_readdir()` will end up calling `VOP_READDIR`, which expects
+	 * a dirent64 structure instead of the `struct dirent` entry parameter.
+	 */
+	struct dirent64 entry64;
```

Comment:
While we're still at it, have declarations grouped in a single block, in a reverse xmas tree manner, i.e.:
```C
        /**
	 * `sys_readdir()` will end up calling `VOP_READDIR`, which expects
	 * a dirent64 structure instead of the `struct dirent` entry parameter.
	 */
        struct dirent64 entry64;
	struct vfscore_file *fp;
	int error;
```